File: templates/ruflet_flutter_template/apple_packages/ruflet_apple/Sources/RufletExtensions/RufletSecureStorage/Sources/Extension.swift
import RufletEngine

@MainActor
public struct RufletSecureStorageExtension: RufletExtension {
  public init() {}

  public var renderedControlTypes: Set<String> { [] }
  public var serviceControlTypes: Set<String> { RufletSecureStorage.controlTypes }

  public func createService(for control: RufletControl) -> RufletService? {
    control.type == "SecureStorage" ? SecureStorageService(control: control) : nil
  }
}

File: templates/ruflet_flutter_template/ruflet_packages/ruflet_secure_storage/lib/src/extension.dart
import 'package:ruflet/ruflet.dart';

import 'secure_storage.dart';

class Extension extends RufletExtension {
  @override
  RufletService? createService(Control control) {
    switch (control.type) {
      case "SecureStorage":
        return SecureStorageService(control: control);
      default:
        return null;
    }
  }
}
